### DISCUSSION TITLE
[Feature]: Is the way to run PIP programmatically?

### DISCUSSION THREAD

**Discussion by ZubairovSalim** [TheWidlarzGroup/react-native-video#4974] [Ideas]
### What problem are you trying to solve?

I am trying to realize custom player controls and I need a PIP button, but I can't find a method for this. :(

### Proposed solution

A way to run PIP (Picture-in-Picture) programmatically, so it can be triggered from a custom PIP button in custom player controls.

**Comment by freeboub** [comment]
If I were you I would try using this PR : https://github.com/TheWidlarzGroup/react-native-video/pull/3385
It enables pip on Android and maybe answer to your question!

**Comment by ZubairovSalim** [comment]
@freeboub Thanks for answer! Is any chance it will be in master within a month?

**Comment by TWG-AI** [comment]
In v7, PiP is fully programmable via the `VideoViewRef`:

```ts
const videoRef = useRef<VideoViewRef>(null);

videoRef.current?.enterPictureInPicture();
videoRef.current?.exitPictureInPicture();
videoRef.current?.canEnterPictureInPicture(); // boolean check
```

Set `pictureInPicture={true}` on `<VideoView>` to enable the PiP button / support.

> I'm an AI — double-check the exact ref setup in the v7 docs.
